src/libs/ReportUtils.ts

```diff
@@ -3756,6 +3762,12 @@ function getReportName(
     personalDetails?: Partial<PersonalDetailsList>,
     invoiceReceiverPolicy?: OnyxEntry<Policy>,
     shouldIncludePolicyName = false,
+    transactions?: OnyxCollection<Transaction>,
+    reports?: OnyxCollection<Report>,
+    draftReports?: OnyxCollection<Report>,
+    reportNameValuePairs?: OnyxCollection<ReportNameValuePairs>,
+    policyTagLists?: OnyxCollection<PolicyTagLists>,
+    policies?: OnyxCollection<Policy>,
```

Comment:
>Frankly I'm surprised that we need some many arguments to generate the name.

Me too 😄 


>If we have to compute the reportName in the frontend codebase, then maybe temporary could somehow split this function for reports and for Search?

I raised this in Slack with the team and didn't get much support since that'd create another fork in the logic. How exactly would you propose the split? Maybe I can bring this up again because I also agree with a separate approach since the refactor might get out of hand.

Reply:
One thing that I was thinking, is that even after this refactor, there's nothing stopping others from directly connecting these methods to Onyx in the future and the whole logic breaking for Search. We should probably think of a way to enforce that once the refactor is done.